front/src/app/components/ThemeSwitcher.tsx

```diff
@@ -1,16 +1,100 @@
 'use client';
 
-import { useTheme } from 'next-themes';
 import { Button } from '@heroui/react';
-import { MoonFilledIcon } from '@heroui/shared-icons';
+import { MoonFilledIcon, SunFilledIcon } from '@heroui/shared-icons';
+import React, { useEffect, useReducer, useState } from 'react';
 
-export function ThemeSwitcher() {
-  const { resolvedTheme, setTheme } = useTheme();
+export const ThemeScript = ({ children }: { children: React.ReactNode }) => {
+  const setSystemTheme = () => {
+    const theme = window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+    const htmlEl = document.documentElement;
+    htmlEl.classList.remove(theme === 'dark' ? 'light' : 'dark');
+    htmlEl.classList.add(theme);
+    htmlEl.style.colorScheme = theme;
+  };
 
   return (
-    <Button isIconOnly onPress={() => setTheme(resolvedTheme === 'light' ? 'dark' : 'light')}
+    <>
+      <script dangerouslySetInnerHTML={{ __html: `(${setSystemTheme.toString()})()` }} />
+      {children}
+    </>
+  );
+};
+
+function setTheme(theme: 'dark' | 'light') {
+  if (typeof window === 'undefined') return;
+
+  const htmlEl = document.documentElement;
+  htmlEl.classList.remove(theme === 'dark' ? 'light' : 'dark');
+  htmlEl.classList.add(theme);
+  htmlEl.style.colorScheme = theme;
+}
+
+const systemWantsDark = (): boolean => {
+  if (typeof window === 'undefined') return false;
+  return window.matchMedia('(prefers-color-scheme: dark)').matches;
+};
+
+const themeReducer = (prevState: themeState, from: 'system' | 'user') => {
```

Comment:
I prefer reducer to have `state` and `action` arguments but not really important
For me an action should have shape `{ type: 'actionName', payload: {...} }` 
Its more declarative, and when you see at usage you can know what the reducer should do with this action

Reply:
Good point. I had the feeling that conventions would call such a name and possibly a typical format.
I made an action type and set the payload/internal action, to only `'toggle'`, which is the only action we support right now.